reframe/core/containers.py

```diff
@@ -119,32 +116,57 @@ class Sarus(ContainerPlatform):
     #: :default: :class:`False`
     with_mpi = fields.TypedField(bool)
 
+    #: Using metahub to fetch instance specific images
+    #:
+    #: :type: boolean
+    #: :default: :class:`False`
+    with_metahub = fields.TypedField(bool)
+    #: Skipping pull alltogether to speed up runs
+    #:
+    #: :type: boolean
+    #: :default: :class:`False`
+    skip_pull = fields.TypedField(bool)
+
     def __init__(self):
         super().__init__()
         self.with_mpi = False
         self._command = 'sarus'
+        self.with_metahub = False
+        self.skip_pull = False
 
     def emit_prepare_commands(self):
         # The format that Sarus uses to call the images is
         # <reposerver>/<user>/<image>:<tag>. If an image was loaded
         # locally from a tar file, the <reposerver> is 'load'.
-        if self.image.startswith('load/'):
+        if self.image.startswith('load/') or self.skip_pull:
             return []
-
-        return [self._command + ' pull %s' % self.image]
+        # Using 
+        if self.with_metahub:
+            tmpdir = tempfile.mkdtemp()
+            return [
+                'docker pull -q mh.qnib.org/'+self.image,
+                'docker save -o '+tmpdir+'/image.tar mh.qnib.org/'+self.image,
```

Comment:
Couldn't `sarus pull` work here? I don't understand very much the purpose of the `save` and `load`.

Reply:
yes, `sarus pull` seems to need a API call that the proxy Metahub is not implementing (yet). working with the SARUS folks to fix that soon(-ish).

Reply:
In this case, you should put this logic on the test as a prerun step. When sarus can pull directly, you just remove the workaround from the particular test.

Reply:
Btw, you can use [skopeo](https://github.com/containers/skopeo) to copy the image directly as a docker-archive and load it with sarus